draco/analysis/flagging.py

```diff
@@ -2431,49 +2443,47 @@ def process(self, data):
                 f"data_stack ({dset_stack.shape})"
             )
 
-        weight_stack = self.data_stack.weight[:]
-        weight = data.weight[:]
+        weight_stack = self.data_stack.weight[:].local_array
+        weight = data.weight[:].local_array
 
         # Find the median offset between the stack and the daily data
         if self.match_median:
             # Find the parts of the both the stack and the daily data that are both
             # measured
-            mask = (
-                ((weight[:] > 0) & (weight_stack[:] > 0))
-                .astype(np.float32)
-                .view(np.ndarray)
-            )
+            mask = ((weight[:] > 0) & (weight_stack[:] > 0)).astype(np.float32)
 
-            # ... get the median of the stack in this common subset
+            # Get the median of the stack in this common subset
             stack_med_real = weighted_median.weighted_median(
-                dset_stack.real.view(np.ndarray).copy(), mask
+                np.ascontiguousarray(dset_stack.real), mask
```

Comment:
Obviously not critical, but just curious: what is the reason for the call to `ascontiguousarray`? Is this related to the change to using `local_array`'s above? Is the stack somehow not in C ordering in this case?

Reply:
It's because we look at the real and imaginary components individually, and those aren't contiguous on their own (you can imagine that storing the full complex dataset in a contiguous format would mean that taking only a component of that would no longer be contiguous). This was being implicitly done by the `.copy()` call, so I just changed that to make the behaviour clearer (without actually changing what's being done)